src/compiler/utilities.ts

```diff
@@ -476,6 +476,38 @@ namespace ts {
         return false;
     }
 
+    /* @internal */
+    export function isDeclarationWithTypeParameters(node: Node): node is DeclarationWithTypeParameters;
+    export function isDeclarationWithTypeParameters(node: DeclarationWithTypeParameters): node is DeclarationWithTypeParameters {

```

Comment:
>export function isDeclarationWithTypeParameters(node: DeclarationWithTypeParameters): node is DeclarationWithTypeParameters { [](start = 4, length = 125)

What is the purpose of the overload here? I think you can just remove it and have one function that takes Node. 

If you have an argument of DeclarationWithTypeParameters type then you already know enough at compile time not to call this function.